lib/standard/math.nit

```diff
@@ -22,9 +22,45 @@ in "C header" `{
 	#include <time.h>
 `}
 
+in "C" `{
+	/* Is rand shortcut? */
+	static int nit_rand_seeded;
+	/* Current rand seed if seeded */
+	static unsigned int nit_rand_seed;
+
+	#define NIT_RAND_MAX 0x7fffffff
+
+	/* This algorithm is mentioned in the ISO C standard, here extended
+	for 32 bits.  */
+	static int
+	nit_rand(void) {
+		unsigned int next = nit_rand_seed;
+		int result;
+
+		next *= 1103515245;
+		next += 12345;
+		result = (unsigned int) (next / 65536) % 2048;
+
+		next *= 1103515245;
+		next += 12345;
+		result <<= 10;
+		result ^= (unsigned int) (next / 65536) % 1024;
+
+		next *= 1103515245;
+		next += 12345;
+		result <<= 10;
+		result ^= (unsigned int) (next / 65536) % 1024;
+
+		nit_rand_seed = next;
+
+		return result;
+	}
+`}
```

Comment:
This rely on precise and reproductible integer arithmetic and overflow. Currently Int in nit are not that reliable.